### DISCUSSION TITLE
How to access to the current XY location and rotation of the agent?

### DISCUSSION THREAD

**Discussion by Berayksl** [PKU-Alignment/safety-gymnasium#126] [Q&A]
I created a custom environment by using the config dictionary as in the example code:

    from __future__ import annotations
    import safety_gymnasium
    
    if __name__ == '__main__':
      config = {
          'lidar_conf.max_dist': 3,
          'lidar_conf.num_bins': 16,
          'placements_conf.extents': [-1.5, -1.5, 1.5, 1.5],
          'Hazards': {
              'num': 1,
              'size': 0.7,
              'locations': [(0, 0)],
              'is_lidar_observed': True,
              'is_constrained': True,
              'keepout': 0.705,
          },
          'Goal': {
              'keepout': 0.305,
              'size': 0.3,
              'locations': [(1.1, 1.1)],
              'is_lidar_observed': True,
          },
      }
    
    # environment from the config
    env = safety_gymnasium.make('SafetyPointGoalBase-v0', config=config)
    env = safety_gymnasium.wrappers.SafetyGymnasium2Gymnasium(env)
    
    s, i = env.reset(seed=42)
    
    for _ in range(1000):
        action = env.action_space.sample()
        s, a, d, t, i = env.step(action)
        env.render()
    print('done')

However, I couldn't figure out how to access to the current XY location and rotation angle of the agent after each action. It was so easy to access to them in Safe Gym library by using env.robot_pos and env.robot_rot but they don't exist in safety gymnasium. I would appreciate it if someone can help me with this. 

Thanks!

**Comment by muchvo** [comment]
You can refer to #54 and #112 .

**Reply by Berayksl** [reply]
Thank you so much!
